### ISSUE TITLE
[Bug]: Controlnet preprocessor leres++ fails

### ISSUE THREAD

**Issue by GodEmperor785** [lllyasviel/stable-diffusion-webui-forge#174]
### Checklist

- [ ] The issue exists after disabling all extensions
- [X] The issue exists on a clean installation of webui
- [X] The issue is caused by an extension, but I believe it is caused by a bug in the webui
- [X] The issue exists in the current version of the webui
- [X] The issue has not been reported before recently
- [ ] The issue has been reported before but has not been fixed yet

### What happened?

When generating image with leres++ preprocessor and controlnet depth there is an error in console and image is generated without controlnet
However when using "allow preview" in controlnet leres++ produces output, issue is only when generating image with input image in controlent and leres++ as preprocessor. It works fine if I manually make preview with leres++, move the result depth image to input image, choose "None" as preprocessor and generate with controlnet depth
From my limited testing it seems that only leres++ has this problem, I successfully generated images with midas, leres (without "++") and depth_anything.
Testing was done on SD1.5 on several models

Also it seems that setting "Path to directory containing annotator model directories (requires restart, overrides corresponding command line flag)" in controlnet section of settings in webui doesn't work. I added a path there to my preprocessors directory which I used in normal A1111 webui but in forge it still downloaded into forge-webui/models/ControlNetPreprocessor instead of using exactly the same files at specified path. Should I make a separate issue for this?

If it matters I installed webui-forge from the "one-click installation package" in readme file (I also ran update.bat after unpacking so forge version should be up to date)

### Steps to reproduce the problem

1. in txt2img enable controlnet, choose depth, preprocessor leres++
2. set input image
3. generate new image with controlnet
4. there is error in console and image is generated without controlnet

### What should have happened?

Image should be generated with controlnet and without error

### What browsers do you use to access the UI ?

Brave

### Sysinfo

[sysinfo-2024-02-10-18-55.json](https://github.com/lllyasviel/stable-diffusion-webui-forge/files/14230220/sysinfo-2024-02-10-18-55.json)

### Console logs

```Shell
2024-02-10 19:52:00,088 - ControlNet - INFO - ControlNet Input Mode: InputMode.SIMPLE██| 20/20 [00:01<00:00, 19.55it/s]
2024-02-10 19:52:00,089 - ControlNet - INFO - Using preprocessor: depth_leres++
2024-02-10 19:52:00,091 - ControlNet - INFO - preprocessor resolution = 512
*** Error running process: H:\AI\webui_forge\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "H:\AI\webui_forge\webui\modules\scripts.py", line 798, in process
        script.process(p, *script_args)
      File "H:\AI\webui_forge\system\python\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "H:\AI\webui_forge\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 526, in process
        self.process_unit_after_click_generate(p, unit, params, *args, **kwargs)
      File "H:\AI\webui_forge\system\python\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "H:\AI\webui_forge\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 303, in process_unit_after_click_generate
        preprocessor_output = preprocessor(
      File "H:\AI\webui_forge\webui\extensions-builtin\forge_legacy_preprocessors\scripts\legacy_preprocessors.py", line 100, in __call__
        result, is_image = self.call_function(img=input_image, res=resolution, thr_a=slider_1, thr_b=slider_2, **kwargs)
      File "H:\AI\webui_forge\webui\extensions-builtin\forge_legacy_preprocessors\legacy_preprocessors\preprocessor.py", line 249, in leres
        result = model_leres(img, thr_a, thr_b, boost=boost)
      File "H:\AI\webui_forge\webui\extensions-builtin\forge_legacy_preprocessors\annotator\leres\__init__.py", line 79, in apply_leres
        depth = estimateboost(input_image, model, 0, pix2pixmodel, max(width, height))
      File "H:\AI\webui_forge\webui\extensions-builtin\forge_legacy_preprocessors\annotator\leres\leres\depthmap.py", line 418, in estimateboost
        whole_estimate = doubleestimate(img, net_receptive_field_size, whole_image_optimal_size, pix2pixsize, model, model_type, pix2pixmodel)
      File "H:\AI\webui_forge\webui\extensions-builtin\forge_legacy_preprocessors\annotator\leres\leres\depthmap.py", line 143, in doubleestimate
        pix2pixmodel.test()
      File "H:\AI\webui_forge\webui\extensions-builtin\forge_legacy_preprocessors\annotator\leres\pix2pix\models\base_model.py", line 106, in test
        self.forward()
      File "H:\AI\webui_forge\webui\extensions-builtin\forge_legacy_preprocessors\annotator\leres\pix2pix\models\pix2pix4depth_model.py", line 116, in forward
        self.fake_B = self.netG(self.real_A)  # G(A)
      File "H:\AI\webui_forge\system\python\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
        return self._call_impl(*args, **kwargs)
      File "H:\AI\webui_forge\system\python\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
        return forward_call(*args, **kwargs)
      File "H:\AI\webui_forge\system\python\lib\site-packages\torch\nn\parallel\data_parallel.py", line 171, in forward
        raise RuntimeError("module must have its parameters and buffers "
    RuntimeError: module must have its parameters and buffers on device cuda:0 (device_ids[0]) but found one of them on device: cpu

---
*** Error running process_before_every_sampling: H:\AI\webui_forge\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "H:\AI\webui_forge\webui\modules\scripts.py", line 830, in process_before_every_sampling
        script.process_before_every_sampling(p, *script_args, **kwargs)
      File "H:\AI\webui_forge\system\python\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "H:\AI\webui_forge\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 533, in process_before_every_sampling
        self.process_unit_before_every_sampling(p, unit, self.current_params[i], *args, **kwargs)
    KeyError: 0

---
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:00<00:00, 20.00it/s]
*** Error running postprocess_batch_list: H:\AI\webui_forge\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "H:\AI\webui_forge\webui\modules\scripts.py", line 854, in postprocess_batch_list
        script.postprocess_batch_list(p, pp, *script_args, **kwargs)
      File "H:\AI\webui_forge\system\python\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "H:\AI\webui_forge\webui\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 539, in postprocess_batch_list
        self.process_unit_after_every_sampling(p, unit, self.current_params[i], pp, *args, **kwargs)
    KeyError: 0

---
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:01<00:00, 17.50it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:01<00:00, 20.20it/s]
```

**Comment by GodEmperor785** [comment]
I tried using leres++ today after updating and in the first run it generated correctly with controlnet, what I did:
1. update.bat and run.bat, wait for forge to start
2. generate one image in txt2img
3. move that image to controlnet input
4. select depth leres++ and enable controlnet
5. image generated fine with controlnet
But on the next try (just doing generate again without changing any settings) it failed with error mentioned in the issue. And after that even trying to run the preprocessor alone (just the preview) it started to fail with first error from the issue:
"RuntimeError: module must have its parameters and buffers on device cuda:0 (device_ids[0]) but found one of them on device: cpu"
Other preprocessors still worked fine.
I tried again after restarting forge and started from making a preview, it also worked fine on the first try but repeating failed with same errors as mentioned
So now it looks like leres++ works only on first run (at least for me)
